src/components/application_manager/src/commands/mobile/system_request.cc

```diff
@@ -53,12 +56,360 @@ uint32_t SystemRequest::index = 0;
+                                      .asString()
+                                      .length() << "] exceeds max length ["
+                               << max_package_name_length_ << "]in json file.");
+              result = false;
+            }
+          }
+        }
+      }
+
+      if (os_type.empty()) {
+        LOG4CXX_WARN(logger_,
+                     error_prefix_
+                         << "Can't find mobile OS type in json file.");
+        result = false;
+        continue;
+      }
+
+      // Languages verification
+      if (!app_data[os_type].keyExists(json::languages)) {
+        LOG4CXX_WARN(logger_, error_prefix_ << "\"languages\" not exists");
+        result = false;
+        continue;
+      }
+      if (!ValidateLanguages(app_data[os_type][json::languages]))
+        result = false;
+    }
+    return result;
+  }
+
+  bool ValidateAppIdAndAppName(
+      const smart_objects::SmartObject& app_data) const {
+    // Verify appid
+    if (!app_data.keyExists(json::appId)) {
+      LOG4CXX_WARN(logger_, error_prefix_ << "Can't find app ID in json file.");
+      return false;
+    }
+    bool result = true;
+    // Verify appid lenght
+    const std::string app_id(app_data[json::appId].asString());
+    if (app_id.length() > max_app_id_length_) {
+      LOG4CXX_WARN(logger_,
+                   error_prefix_
+                       << "An Object ID length exceeds maximum allowed ["
+                       << app_id.length() << "]>[" << max_app_id_length_
+                       << "]");
+      result = false;
+    }
+
+    // Verify that appid is unique
+    if (app_ids_map_.find(app_id) != app_ids_map_.end()) {
+      LOG4CXX_WARN(logger_,
+                   error_prefix_ << "An Object ID is not unigue [" << app_id
+                                 << "]");
+      result = false;
+    } else {
+      app_ids_map_[app_id] = 1;
+    }
+
+    // Verify that app is not registered yet
+    ApplicationSharedPtr registered_app =
+        ApplicationManagerImpl::instance()->application_by_policy_id(app_id);
+    if (registered_app) {
+      LOG4CXX_DEBUG(logger_,
+                    error_prefix_ << "Application with the same id: " << app_id
+                                  << " is registered already.");
+      result = false;
+    }
+    // Verify app name exist
+    if (!app_data.keyExists(json::name)) {
+      LOG4CXX_WARN(logger_,
+                   error_prefix_ << "Can't find app name in json file.");
+      return false;
+    }
+    // And app name length
+    const std::string appName(
+        app_data[json::name].asString());
+    if (appName.length() > max_app_name_length_) {
+      LOG4CXX_WARN(logger_,
+                   error_prefix_
+                       << "Name of application exceeds maximum allowed ["
+                       << appName.length() << "]>[" << max_app_name_length_
+                       << "].");
+      result = false;
```

Comment:
@okoreniak, imagine there will be additional checks later on. According the check before we should break validation, but current code assumes that this is last check in the method..